### ISSUE TITLE
Switch and Checkbox elements are mistakingly manipulated on label click

### ISSUE THREAD

**Issue by robskaar** [ant-design/ant-design#47031]
### Reproduction link

[![Edit on CodeSandbox](https://codesandbox.io/static/img/play-codesandbox.svg)](https://codesandbox.io/p/sandbox/form-disabled-antd-5-13-1-forked-8p4prv?file=%2Fdemo.tsx%3A20%2C12)

### Steps to reproduce

When clicking the label of a Form.item component that contains a Switch/Checkbox with a name attribute set, the Switch/Checkbox state is modified, in other input components, e.g. a select/text, the component gains focus instead.
This leads to different mental models of the user trying to use a input component, and might lead to unintentional toggles of Switch/Checkbox when either scrolling down the page, or when trying to focus the input. 
Steps:

1. add following component:
        <Form.Item label="Switch" name="switchup" valuePropName="checked">
          <Switch />
        </Form.Item>
2. Click the label
3. the state of the toggle (Switch/Checkbox) is now changed
4. remove name attribute on Form.Item
5. click the label
6. the state is no longer modified when name is not present on the input componenet.

Issue happens when property name is set, if name property is removed issue is gone.
i assume it wants to move focus to the component on label click, instead it actually toggles/Detoggle.

### What is expected?

input element is focused like all other input elements in the package

### What is actually happening?

input elements ( checkbox & switch) has their current state modified instead of just gaining focus. this can cause unintentional toggles from a user expecting the onclick of the label to behave like the other input elements.

| Environment | Info |
| --- | --- |
| antd | 5.13.1 |
| React | ^18.0 |
| System | windows |
| Browser | firefox |

**Comment by afc163** [comment]
It is intented, see https://github.com/ant-design/ant-design/issues/43175#issuecomment-1605926864

**Comment by robskaar** [comment]
Hi, as i'm not familiar with the language in #43175 it didn't show on my searches for the problem i had.

i just found out that it is indeed intended, and apparently my mental model is not the norm, i'd expect an even behavior, not one input type being focused, and another type being modified.

But here is a a link here to MDN explaining the reason behind it for any future readers who also get's this issue and dont understand the language in the linked post:
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/label#for

TLDR: When a user clicks or touches/taps a label, the browser passes the focus to its associated input (the resulting event is also raised for the input). That increased hit area for focusing the input provides an advantage to anyone trying to activate it — including those using a touch-screen device.

**Comment by afc163** [comment]
Check https://codepen.io/afc163/pen/MWzbRGe for native behavior.

**Comment by robskaar** [comment]
> Check https://codepen.io/afc163/pen/MWzbRGe for native behavior.

Yes as posted i found that its native behavior once linking label to input, i think the use case might still be an acceptable issue - i certainly would love the option to modify this to fit my needs.

 so opening it as a feature suggestion, i don't see why it shouldnt be possible to set the "labelClickBehavior" between "focus/Toggle" on switch and checkbox elements
